src/fns.c

```diff
@@ -3154,49 +3154,7 @@ into shorter lines.  */)
   return make_number (encoded_length);
 }
 
-DEFUN ("base64-encode-string", Fbase64_encode_string, Sbase64_encode_string,
-       1, 2, 0,
-       doc: /* Base64-encode STRING and return the result.
-Optional second argument NO-LINE-BREAK means do not break long lines
-into shorter lines.  */)
-  (Lisp_Object string, Lisp_Object no_line_break)
-{
-  ptrdiff_t allength, length, encoded_length;
-  char *encoded;
-  Lisp_Object encoded_string;
-  USE_SAFE_ALLOCA;
-
-  CHECK_STRING (string);
-
-  /* We need to allocate enough room for encoding the text.
-     We need 33 1/3% more space, plus a newline every 76
-     characters, and then we round up. */
-  length = SBYTES (string);
-  allength = length + length/3 + 1;
-  allength += allength / MIME_LINE_LENGTH + 1 + 6;
-
-  /* We need to allocate enough room for decoding the text. */
-  encoded = SAFE_ALLOCA (allength);
-
-  encoded_length = base64_encode_1 (SSDATA (string),
-				    encoded, length, NILP (no_line_break),
-				    STRING_MULTIBYTE (string));
-  if (encoded_length > allength)
-    emacs_abort ();
-
-  if (encoded_length < 0)
-    {
-      /* The encoding wasn't possible. */
-      error ("Multibyte character in data for base64 encoding");
```

Comment:
You don't seem to be checking the encoded length, or handling the multibyte case in your Rust implementation. What happens if there's a multibyte character? I suspect we need the same error in the same situation.